enhancements/scheduling/prevent-user-workloads-from-being-scheduled-on-control-plane-nodes.md

```diff
@@ -0,0 +1,412 @@
+
+### Expected Outcomes
+
+- Improved protection of critical cluster resources, ensuring that control plane nodes and nodes with 
+ resources are shielded from inappropriate workloads.
+- Increased flexibility for administrators to tailor scheduling policies to the specific needs of their organization and operational environment.
+
+
+
+### Workflow Description
+
+<!-- Explain how the user will use the feature. Be detailed and explicit.
+Describe all of the actors, their roles, and the APIs or interfaces
+involved. Define a starting state and then list the steps that the
+user would need to go through to trigger the feature described in the
+enhancement. Optionally add a
+[mermaid](https://github.com/mermaid-js/mermaid#readme) sequence
+diagram.
+
+Use sub-sections to explain variations, such as for error handling,
+failure recovery, or alternative outcomes.
+
+For example:
+
+**cluster creator** is a human user responsible for deploying a
+cluster.
+
+**application administrator** is a human user responsible for
+deploying an application in a cluster.
+
+1. The cluster creator sits down at their keyboard...
+2. ...
+3. The cluster creator sees that their cluster is ready to receive
+   applications, and gives the application administrator their
+   credentials. -->
+   
+This section outlines the workflow for enforcing the proposed mechanism for prevention of non-control plane workloads from being scheduled on control plane nodes and allow for exceptions based on administrative configurations.
+
+#### Actors
+
+Pre-Workflow:
+- OpenShift Engineers: Ensure all control plane and essential workloads ship with the correct tolerations out of the box to seamlessly operate on NoExecute tainted nodes. This preparation is crucial for the uninterrupted functioning of OpenShift's core services.
+
+Workflow:
+- Cluster Administrator: Tasked with applying NoExecute taints to control plane nodes or nodes belonging to other specialized groups. For extending this approach to other special node groups, they must coordinate with relevant teams to ensure those workloads include necessary tolerations. They are also responsible for creating and applying ValidatingAdmissionPolicy and ValidatingAdmissionPolicyBinding resources to configure which taint tolerations are disallowed for pods, except in specific namespaces designated by a particular label (e.g., openshift.io/control-plane-namespace, or another custom label). This role involves a strategic overview of the cluster's security and workload management policies.
+- Namespace Administrator: Manages their namespaces, including labeling them appropriately (e.g., with openshift.io/control-plane-namespace, or another custom label) when exemptions to the default scheduling policies are needed. This role requires understanding the impact of these labels on workload scheduling and compliance with the cluster's security policies.
```

Comment:
Open to alternatives for "openshift.io/control-plane-namespace" key. E.g.:
- openshift.io/control-plane-workload
- openshift.io/control-plane-component
- openshift.io/control-plane-group

Reply:
Do you mean we should mention other alternatives in the text? Or that we should choose one of those now to recommend?

Reply:
The latter

Reply:
maybe just openshift.io/control-plane

Reply:
I decided for openshift.io/control-plane simpler label, feel free to re-open the discussion

Reply:
`openshift.io/control-plane` ++